### ISSUE TITLE
[bitnami/kubernetes-event-exporter] Observed a panic: "invalid memory address or nil pointer dereference"

### ISSUE THREAD

**Issue by sherifkayad** [bitnami/charts#17961]
### Name and Version

bitnami/kubernetes-event-exporter (chart version 2.4.7)

### What architecture are you using?

amd64

### What steps will reproduce the bug?

1. Install the Chart to a Kubernetes cluster with version 1.27
2. The Pod crashes with the error below:
```
{"level":"info","time":"2023-07-26T14:33:37Z","message":"Reading config file /data/config.yaml"}
{"level":"info","time":"2023-07-26T14:33:37Z","message":"set config.maxEventAgeSeconds=5 (default)"}
{"level":"warn","time":"2023-07-26T14:33:37Z","message":"metrics name prefix is empty, setting config.metricsNamePrefix='event_exporter_' is recommended"}
{"level":"info","name":"dump","type":"*sinks.File","time":"2023-07-26T14:33:37Z","message":"Registering sink"}
E0726 14:34:01.491739       1 runtime.go:79] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 49 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x27edca0?, 0x4958310})
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:75 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc0008d6598?})
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:49 +0x75
panic({0x27edca0, 0x4958310})
	/usr/local/go-1.20.6/src/runtime/panic.go:884 +0x213
k8s.io/client-go/discovery.convertAPIResource(...)
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/client-go/discovery/aggregated_discovery.go:88
k8s.io/client-go/discovery.convertAPIGroup({{{0x0, 0x0}, {0x0, 0x0}}, {{0xc0004d7038, 0x15}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/client-go/discovery/aggregated_discovery.go:69 +0x570
k8s.io/client-go/discovery.SplitGroupsAndResources({{{0xc000121e90, 0x15}, {0xc00088e2c0, 0x1b}}, {{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/client-go/discovery/aggregated_discovery.go:35 +0x118
k8s.io/client-go/discovery.(*DiscoveryClient).downloadAPIs(0x0?)
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/client-go/discovery/discovery_client.go:310 +0x47c
k8s.io/client-go/discovery.(*DiscoveryClient).GroupsAndMaybeResources(0x0?)
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/client-go/discovery/discovery_client.go:198 +0x5c
k8s.io/client-go/discovery.ServerGroupsAndResources({0x31a24b0, 0xc000691140})
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/client-go/discovery/discovery_client.go:392 +0x59
k8s.io/client-go/discovery.(*DiscoveryClient).ServerGroupsAndResources.func1()
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/client-go/discovery/discovery_client.go:356 +0x25
k8s.io/client-go/discovery.withRetries(0x2, 0xc0008d7690)
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/client-go/discovery/discovery_client.go:621 +0x71
k8s.io/client-go/discovery.(*DiscoveryClient).ServerGroupsAndResources(0x0?)
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/client-go/discovery/discovery_client.go:355 +0x3a
k8s.io/client-go/restmapper.GetAPIGroupResources({0x31a24b0?, 0xc000691140?})
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/client-go/restmapper/discovery.go:148 +0x42
github.com/resmoio/kubernetes-event-exporter/pkg/kube.GetObject(0xc00019c388, 0xc000102340, {0x3178680, 0xc0003a5de0})
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/pkg/kube/objects.go:28 +0x108
github.com/resmoio/kubernetes-event-exporter/pkg/kube.(*LabelCache).GetLabelsWithCache(0xc0006890c0, 0xc00019c388)
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/pkg/kube/labels.go:42 +0xab
github.com/resmoio/kubernetes-event-exporter/pkg/kube.(*EventWatcher).onEvent(0xc000689dc0, 0xc00019c280)
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/pkg/kube/watcher.go:103 +0x21d
github.com/resmoio/kubernetes-event-exporter/pkg/kube.(*EventWatcher).OnAdd(0x7f35ab2d3698?, {0x2bf8f00?, 0xc00019c280?})
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/pkg/kube/watcher.go:54 +0x32
k8s.io/client-go/tools/cache.(*processorListener).run.func1()
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/client-go/tools/cache/shared_informer.go:911 +0x134
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x30?)
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:157 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000707f38?, {0x3178640, 0xc000462000}, 0x1, 0xc000460000)
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:158 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x0?, 0x3b9aca00, 0x0, 0x0?, 0x0?)
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:135 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(...)
```

### What is the expected behavior?

The pod runs fine like with previous Kubernetes versions

### What do you see instead?

```
{"level":"info","time":"2023-07-26T14:33:37Z","message":"Reading config file /data/config.yaml"}
{"level":"info","time":"2023-07-26T14:33:37Z","message":"set config.maxEventAgeSeconds=5 (default)"}
{"level":"warn","time":"2023-07-26T14:33:37Z","message":"metrics name prefix is empty, setting config.metricsNamePrefix='event_exporter_' is recommended"}
{"level":"info","name":"dump","type":"*sinks.File","time":"2023-07-26T14:33:37Z","message":"Registering sink"}
E0726 14:34:01.491739       1 runtime.go:79] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 49 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x27edca0?, 0x4958310})
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:75 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc0008d6598?})
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:49 +0x75
panic({0x27edca0, 0x4958310})
	/usr/local/go-1.20.6/src/runtime/panic.go:884 +0x213
k8s.io/client-go/discovery.convertAPIResource(...)
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/client-go/discovery/aggregated_discovery.go:88
k8s.io/client-go/discovery.convertAPIGroup({{{0x0, 0x0}, {0x0, 0x0}}, {{0xc0004d7038, 0x15}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/client-go/discovery/aggregated_discovery.go:69 +0x570
k8s.io/client-go/discovery.SplitGroupsAndResources({{{0xc000121e90, 0x15}, {0xc00088e2c0, 0x1b}}, {{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/client-go/discovery/aggregated_discovery.go:35 +0x118
k8s.io/client-go/discovery.(*DiscoveryClient).downloadAPIs(0x0?)
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/client-go/discovery/discovery_client.go:310 +0x47c
k8s.io/client-go/discovery.(*DiscoveryClient).GroupsAndMaybeResources(0x0?)
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/client-go/discovery/discovery_client.go:198 +0x5c
k8s.io/client-go/discovery.ServerGroupsAndResources({0x31a24b0, 0xc000691140})
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/client-go/discovery/discovery_client.go:392 +0x59
k8s.io/client-go/discovery.(*DiscoveryClient).ServerGroupsAndResources.func1()
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/client-go/discovery/discovery_client.go:356 +0x25
k8s.io/client-go/discovery.withRetries(0x2, 0xc0008d7690)
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/client-go/discovery/discovery_client.go:621 +0x71
k8s.io/client-go/discovery.(*DiscoveryClient).ServerGroupsAndResources(0x0?)
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/client-go/discovery/discovery_client.go:355 +0x3a
k8s.io/client-go/restmapper.GetAPIGroupResources({0x31a24b0?, 0xc000691140?})
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/client-go/restmapper/discovery.go:148 +0x42
github.com/resmoio/kubernetes-event-exporter/pkg/kube.GetObject(0xc00019c388, 0xc000102340, {0x3178680, 0xc0003a5de0})
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/pkg/kube/objects.go:28 +0x108
github.com/resmoio/kubernetes-event-exporter/pkg/kube.(*LabelCache).GetLabelsWithCache(0xc0006890c0, 0xc00019c388)
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/pkg/kube/labels.go:42 +0xab
github.com/resmoio/kubernetes-event-exporter/pkg/kube.(*EventWatcher).onEvent(0xc000689dc0, 0xc00019c280)
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/pkg/kube/watcher.go:103 +0x21d
github.com/resmoio/kubernetes-event-exporter/pkg/kube.(*EventWatcher).OnAdd(0x7f35ab2d3698?, {0x2bf8f00?, 0xc00019c280?})
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/pkg/kube/watcher.go:54 +0x32
k8s.io/client-go/tools/cache.(*processorListener).run.func1()
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/client-go/tools/cache/shared_informer.go:911 +0x134
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x30?)
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:157 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000707f38?, {0x3178640, 0xc000462000}, 0x1, 0xc000460000)
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:158 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x0?, 0x3b9aca00, 0x0, 0x0?, 0x0?)
	/bitnami/blacksmith-sandox/kubernetes-event-exporter-1.2.0/src/github.com/opsgenie/kubernetes-event-exporter/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:135 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(...)
```

**Comment by sherifkayad** [comment]
Probably related to https://github.com/resmoio/kubernetes-event-exporter/issues/91

**Comment by sherifkayad** [comment]
I guess after some readings in tons of issues, I came down to the root cause; that's a weird combination of Kubernetes 1.27 + Prometheus Adapter pushing metrics in a wrong format to the k8s API + a faulty Go k8s API Lib version ..

The last piece of the puzzle is the Prometheus Adapter (https://github.com/kubernetes-sigs/prometheus-adapter/issues/581), luckily the folks issued a fixed release yesterday (`v0.11.0`) and I can confirm that upon upgrading to that one everything is back to normal!

I am closing this issue as completed!
